### PR TITLE
fix(ci): re-enable CodeQL for C++ COMPASS-7540

### PR DESCRIPTION
**Pull request by addaleax**

### CODE REVIEW

File: resources/main-template.cc [hunk 1]
```diff
@@ -35,7 +35,7 @@ using namespace v8;
 #endif
 
 // Snapshot config is supported since https://github.com/nodejs/node/pull/50453
-#if NODE_VERSION_AT_LEAST(21, 6, 0) && !defined(BOXEDNODE_SNAPSHOT_CONFIG_FLAGS)
+#if NODE_VERSION_AT_LEAST(20, 12, 0) && !defined(BOXEDNODE_SNAPSHOT_CONFIG_FLAGS)
```
Comment (addaleax) [inline resources/main-template.cc:38]: drive-by fix, this just happened to get backported into the most recent Node.js 20 release

**Review by mcasimir — APPROVED** [review]
